src/lib/components/login.svelte

```diff
@@ -0,0 +1,94 @@
+<script lang="ts">
+	import Button from '$lib/components/button.svelte'
+	import Input from '$lib/components/input/input.svelte'
+	import adapter from '$lib/adapters'
+	import Typography from '$lib/components/typography.svelte'
+	import type { Advisor } from '$lib/types'
+	import type { ZodFormattedError } from 'zod'
+	import { loginFormSchema } from '$lib/schemas'
+
+	//eslint-disable-next-line @typescript-eslint/no-unused-vars
+	let { registration = $bindable() } = $props()
+	let email = $state('')
+	let password = $state('')
+	let error = $state('')
+	let formErrors: ZodFormattedError<Omit<Advisor, 'confirmPassword'>> | undefined =
+		$state(undefined)
+	let formValid = $state(false)
+	async function login() {
+		try {
+			await adapter.signIn(email, password)
+			email = ''
+			password = ''
+		} catch (e) {
+			error = (e as Error).message
+		}
+	}
+	$effect(() => {
+		const res = loginFormSchema.safeParse({ email, password })
+		if (res.success) {
+			formErrors = undefined
+			formValid = true
+		} else {
+			formErrors = res.error.format()
+			formValid = false
+		}
+	})
+</script>
+
+<div class="login">
+	<div class="login-form">
+		<Input
+			bind:value={email}
+			label="Email"
+			type="email"
+			error={formErrors?.email?._errors ?? undefined}
+		></Input>
+		<Input
+			bind:value={password}
+			type="password"
+			label="Password"
+			error={formErrors?.password?._errors ?? undefined}
```

Comment:
This only makes sense if `_errors` is `null` for some reason. If that's not case it's best to omit it.
```suggestion
			error={formErrors?.password?._errors}
```